File: longestConsecutive.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length <2) return nums.length;
        int largetSeq = 0;
        HashSet<Integer> set = new HashSet();
        for(int num: nums){
            set.add(num);
        }
        
        for(int num: nums){
            if(!set.contains(num-1)){
                int currSeq = 1;
                while(set.contains(num+1)){
                    num++;
                    currSeq++;
                }
                largetSeq = Math.max(largetSeq, currSeq);
            }
            
        }
        return largetSeq;
    }
}

File: Arrays and Hashing/longestConsecutive.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        unique = set()
        longestSeq = 0

        for num in nums:
            unique.add(num)
        
        for num in nums:
            if num-1 not in unique:
                currSeq=1
                while num+1 in unique:
                    num += 1
                    currSeq += 1
                longestSeq = max(longestSeq, currSeq)
        return longestSeq
